webextension/js/popup.js

```diff
@@ -85,13 +85,27 @@ const Logic = {
     });
   },
 
+  userContextId(cookieStoreId = "") {
+    const userContextId = cookieStoreId.replace("firefox-container-", "");
+    return (userContextId !== cookieStoreId) ? Number(userContextId) : false;
+  },
+
   refreshIdentities() {
-    return browser.runtime.sendMessage({
-      method: "queryIdentities"
-    })
-    .then(identities => {
-      this._identities = identities;
-    });
+    return Promise.all([
+      browser.contextualIdentities.query({}),
+      browser.runtime.sendMessage({
```

Comment:
This code merges the two together for now, just so we can reduce the code later easier whilst safely removing our dependency on it here. As mentioned earlier this can be cleared up in a follow up when we have hide and sort here.